### ISSUE TITLE
Error in cutree(dendro, h = heightcutoff) when exp2gcn is ran

### ISSUE THREAD

**Issue by HelloYiHan** [almeidasilvaf/BioNERO#3]
Hi Guys,

BioNERO is really a helpful tool!

When I run the command:
```
final_exp <- exp_preprocess(data, min_exp = 1, variance_filter = TRUE, n = 6500)
net <- exp2gcn(final_exp, SFTpower = power, cor_method = "pearson")
```
('data' is a gene expression data frame with genes in row names and samples in column names), the following error message arises:
```
Error in cutree(dendro, h = heightcutoff) : 
  the 'height' component of 'tree' is not sorted (increasingly)
```
I was wondering how to solve it.

Thank you for your kind help.

Yi

**Comment by almeidasilvaf** [comment]
Hi, @HelloYiHan 

Thank you for your feedback!

That's odd, I've never seen it before. Based on the information you gave, I'd check 2 things:

- convert `data` to a matrix instead of a data frame. That would be:

```
data <- as.matrix(data)
```

- check that your object `power` is a numeric scalar (e.g., 10). The function `SFT_fit()` returns a list with elements `power` (numeric scalar with the power) and `plot` (ggplot object). You might be giving this list as an argument to the `SFTpower` parameter of `exp2gcn()`. If your object `power` is the output of `SFT_fit()`, you should give `power$power`.

Best,
Fabricio

**Comment by HelloYiHan** [comment]
Hi Fabricio,

Thank you for your quick reply.

I have tested my code and I find the reason for this error is that the `min_exp` was set to 1 in the `exp_preprocess` function.  If the `min_exp` was set to more than 1, there will be no error.

Thank you!
Yi

**Comment by almeidasilvaf** [comment]
Glad to know you solved it. :)

I'll close this issue, then.

**Comment by almeidasilvaf** [comment]
Short update on this issue: another user told me he had the same issue. Upon inspection, I found that it was due to small rounding differences in R, which caused problems with the `cutree()` function (see [here](https://stackoverflow.com/questions/46749829/the-height-component-of-tree-is-not-sorted-error-in-cutree)).

This issue is now solved in commit cc31fee5698cb0a604f871996b753ba00fa76e58.

Best, 
Fabricio
